### ISSUE TITLE
Can not parse non-standard boolean values ('yes','no','1','0)

### ISSUE THREAD

**Issue by johanmynhardt** [Afrozaar/wp-api-v2-client-java#16]
**TL;DR: The issue is bad/inconsistent data.**

**In Depth**
The WP-API and our Java Client agrees on the data structure being used to transmit data, which is JSON. The user capabilities is a map of {string: boolean} values, and in JSON it looks something like the following:

    {
      "capabilities": {
        "foo": true,
        "bar": false
      }
    }
    
but we received something like:

    {
      "capabilities": {
        "foo": "1",
        "bar": false
      }
    }

Which can not be deserialized properly and causes the stack trace included in the description.
By digging around in a local wordpress installation's database structure it is found that for complex structures, the entire structure is serialized using PHP's `string serialize(mixed $value)` function.

**wp_usermeta**
In the `wp_usermeta` table an entry with the key `wp_capabilities` is found containing such a serialized value.
Standard format for admin user's `wp_capabilities` value

    mysql> select * from wp_usermeta where meta_key = 'wp_capabilities' and user_id = 1;
    +----------+---------+-----------------+---------------------------------+
    | umeta_id | user_id | meta_key        | meta_value                      |
    +----------+---------+-----------------+---------------------------------+
    |       10 |       1 | wp_capabilities | a:1:{s:13:"administrator";b:1;} |
    +----------+---------+-----------------+---------------------------------+
    
The structure `a:1:{s:13:"administrator";b:1;}` translate to an array of length 1 containing a string-key (of 13 characters, administrator) and a boolean-value, 1, which denotes true.
Changing this value to `a:1:{s:13:"administrator";s:1:"1";}` (array of length 1 containing a string-key of 13 characters, administrator and a string-value of 1 character with value 1)
and making a call to the endpoint exerts the breaking behaviour experienced.
So it appears that someone modified this value, or it was incorrectly modified by a plugin (this is hard to do by hand and getting it wrong will brick some usability if not the wordpress site).

**Solution**

The appropriate solution would be to fix this so that the intended data is actually flowing through WordPress.
